### PR TITLE
paginate cache sets in groups of 2000

### PR DESCRIPTION
**Pull request by tekhaus**
This update should allow shops with very large inventories to successfully complete the cache setting, avoiding the 256kb size limit for a single cache value.

### CODE REVIEW

**Comment by tekhaus** [comment]
[Extend "Accept a maximum number of orders per day" for multi-page caching](https://trello.com/c/bkBHAb7i/39-extend-accept-a-maximum-number-of-orders-per-day-for-multi-page-caching)

File: docs/accept-a-maximum-number-of-orders-per-day/script.liquid [hunk 1]
```diff
@@ -216,7 +212,25 @@
       {% endfor %}
 
       {% if options.restore_inventory_levels_the_next_day__boolean or options.restore_inventory_levels_on_demand__boolean %}
-        {% action "cache", "set", cache_key, inventory_levels_to_zero %}
+        {% comment %}
+          -- NOTE: large inventories can run up against the 256kb Mechanic cache limit, so need to paginate (2000 entries ~200kb)
```
Comment (isaacbowen) [inline docs/accept-a-maximum-number-of-orders-per-day/script.liquid:216]: ❤️

File: docs/accept-a-maximum-number-of-orders-per-day/script.liquid [hunk 2]
```diff
@@ -279,7 +307,9 @@
     {% endfor %}
 
     {% if inventory_levels_to_restore != blank %}
-      {% action "cache", "del", cache_key %}
+      {% for cache_key in cache_keys %}
+        {% action "cache", "del", cache_key %}
+      {% endfor %}
```
Comment (isaacbowen) [inline docs/accept-a-maximum-number-of-orders-per-day/script.liquid:312]: I _think_ there's a hole here, in the event that multiple cache pages are saved, but _afterwards_ the total number of inventory items drops below 2000. In that event, there wouldn't be evidence that multiple cache pages exist, which means that subsequent pages wouldn't be deleted.

Those cache pages will expire, and the time-based cache key means they wouldn't ever be used, sooo I'm fine with any of the following:

1. Solve it with documentation, noting that there's a hole and it's not an issue.
2. Use setex with the cache action, and have each cached document last .. what, 48hrs? Feels risky, though.
3. Look up the cache pages before deleting them, and follow the linked list to delete all linked cache pages.

Let me know if this makes sense, and also let me know if I'm completely off-base about this being an issue. :D

Comment (tekhaus) [reply docs/accept-a-maximum-number-of-orders-per-day/script.liquid:312]: The cache delete is only during the restore loop, so it _should_ always get all of them.

I guess there could be a straggler if someone disabled the restore options _after_ the cache was set. But then, on the next cache set, the last group cache (assuming less inventory items) wouldn't have a "next_key".

Comment (isaacbowen) [reply docs/accept-a-maximum-number-of-orders-per-day/script.liquid:312]: Ahhh, okay, that is a more specific situation than I was thinking. Thanks for spelling it out. :) Am good with this then.

**Pull request merged**